pkg/repository/git.go

```diff
@@ -33,29 +34,47 @@ func newGitRepoController() gitRepoController {
 	}
 }
 
+type cleanStatus struct {
+	IsClean bool
+	Err     error
+}
+
 // IsClean checks if the repository have changes over the commit
 func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Second)
```

Comment:
the new logic keeps having a timeout for safety but the local execution of `git status` should be fairly reliable, however in case the bin of `git` is not installed on host, we fall back on the old logic using `git-go`. In that case, it will be slower (I've seen it take over 20 seconds). In that case the timeout will be useful.